cpu/lpc2387/periph/spi.c

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_lpc2387
+ * @{
+ *
+ * @file
+ * @brief       Low-level SPI driver implementation
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "mutex.h"
+#include "periph/gpio.h"
+#include "periph/spi.h"
+#include "periph_conf.h"
+#include "board.h"
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#if SPI_0_EN
+
+#define SPI_TX_EMPTY                (SSP0SR & SSPSR_TFE)
+#define SPI_BUSY                    (SSP0SR & SSPSR_BSY)
+#define SPI_RX_AVAIL                (SSP0SR & SSPSR_RNE)
+
+/**
+ * @brief Array holding one pre-initialized mutex for each SPI device
+ */
+static mutex_t locks[] =  {
+#if SPI_0_EN
+    [SPI_0] = MUTEX_INIT,
+#endif
+#if SPI_1_EN
+    [SPI_1] = MUTEX_INIT,
+#endif
+#if SPI_2_EN
+    [SPI_2] = MUTEX_INIT
+#endif
+};
+
+int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
```

Comment:
would be cool to put a TODO in here somewhere, that you don't set clock polarity and phase at the moment